### PR TITLE
Texture2D FromStream tests and message

### PR DESCRIPTION
**Pull request by Tzenchor**
First pass at adding a tests to `Texture2D.FromStream` as commented in #4479.
Enveloped `Texture2D.FromStream` in a try-catch block, inside the catch I am throwing a new NotSupportedException with the original one inside and the message "Image type not supported"
Add test class `Texture2DNonVisual` to test project with tests for filetypes: bmp, dds, gif, jpg, png, tiff and tga. All assets reused from other tests.

I have tested this in Windows with the Windows solution and adapted the tests to it (only the tga image provokes an exception), let's see if the other platforms have the same capabilities.

### CODE REVIEW

**Comment by tomspilman** [comment]
Good first pass here @Tzenchor .

You should also add the `Texture2DNonVisualTest.cs` to the XNA project...

 https://github.com/mono/MonoGame/blob/develop/Test/MonoGame.Tests.XNA.csproj

You can then run the test under XNA and verify we get the same results as XNA does.

**Comment by Tzenchor** [comment]
I have added the test class to the XNA test project, but i can't run it as i don't the tooling installed.
Also i have modified the tests to check that the created `Texture2D` at least has the same size than the original image.

I don't know if i can add anything more here apart from checking some pixels.

**Comment by tomspilman** [comment]
> I have added the test class to the XNA test project, but i can't run it as i don't the tooling installed.

Ah... we need to find someone to run that for you and verify we're getting the same output with XNA.

**Comment by Tzenchor** [comment]
I have just added a new test for a dds file that does not contain a texture (in this case it contains a cubemap).
Also have added documentation comments to `Texture2D.cs`, please review them for typos or inaccuracies

**Comment by KonajuGames** [comment]
It would appear that XNA throws an `InvalidOperationException` rather than `NotSupportedException`.
http://xboxforums.create.msdn.com/forums/t/79565.aspx
I don't have XNA installed either, so I went hunting.

**Comment by Tzenchor** [comment]
> It would appear that XNA throws an InvalidOperationException rather than NotSupportedException.

Good to know, i'll update later after finishing some changes for another pull.
Also while investigating further I found this other jewel [XNA Texture2D.FromStream documentation](https://msdn.microsoft.com/en-us/library/ff434103.aspx?f=255&MSPPError=-2147217396)
It appears that XNA can only load .gif, .png and .jpg from a Stream

**Comment by KonajuGames** [comment]
Correct in that XNA only supported those three formats. Supporting more without requiring a dependency on a large image loading library isn't a problem though. Think of GIF, PNG and JPG as the minimum set, and if we can support other common formats without adding a large amount of code, then there is little reason not to.

**Comment by tomspilman** [comment]
@Tzenchor - FYI...  i'm switching out to a new build server, so the errors you are seeing might not be your fault.   I'll let you know when I'm done and if I need fixes from you.

**Comment by tomspilman** [comment]
@Tzenchor - Ok... the build server is working and things were fine with your changes.

**Comment by Tzenchor** [comment]
This morning I run the tests against XNA and all tests passed correctly after adding the missing assets to the XNA project

**Comment by KonajuGames** [comment]
Excellent. I'll merge this now.

File: MonoGame.Framework/Graphics/Texture2D.cs [hunk 1]
```diff
@@ -27,28 +29,58 @@ public Rectangle Bounds
...
+        /// <param name="graphicsDevice"></param>
+        /// <param name="width"></param>
+        /// <param name="height"></param>
         public Texture2D(GraphicsDevice graphicsDevice, int width, int height)
             : this(graphicsDevice, width, height, false, SurfaceFormat.Color, SurfaceType.Texture, false, 1)
         {
         }
-
+        /// <summary>
+        /// Creates a new texture of a given size with a surface format and optional mipmaps
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Texture2D.cs]: @Tzenchor - In the future remember to put periods at the end of the sentence in the reference docs.  I didn't get time to review this PR before it was merged else I would have caught this then.

**Pull request merged**